src/utils.ts

```diff
@@ -410,19 +422,18 @@ export function addToMessageList<T extends FormatMessageResponse>(
     sortDirection: 'ascending',
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     selectValueToCompare: (m) => m[sortBy]!.getTime(),
+    selectKey: (m) => m.id,
   });
 
   // message already exists and not filtered with timestampChanged, update and return
-  if (!timestampChanged && newMessage.id) {
-    if (newMessages[insertionIndex] && newMessage.id === newMessages[insertionIndex].id) {
-      newMessages[insertionIndex] = newMessage;
-      return newMessages;
-    }
-
-    if (newMessages[insertionIndex - 1] && newMessage.id === newMessages[insertionIndex - 1].id) {
-      newMessages[insertionIndex - 1] = newMessage;
-      return newMessages;
-    }
```

Comment:
We no longer need to check this case: if a message needs to be updated, `insertionIndex` is guaranteed to point at it.